pkg/services/rpcsrv/server_test.go

```diff
@@ -2275,79 +2275,173 @@ func TestSubmitOracle(t *testing.T) {
+		t.Run("invalid request bytes", submitNotaryRequest(t, true, neorpc.InvalidParamsCode, `"not-a-request"`))
+		t.Run("invalid request", func(t *testing.T) {
+			mainTx := &transaction.Transaction{
+				Attributes:      []transaction.Attribute{{Type: transaction.NotaryAssistedT, Value: &transaction.NotaryAssisted{NKeys: 1}}},
+				Script:          []byte{byte(opcode.RET)},
+				ValidUntilBlock: 123,
+				Signers:         []transaction.Signer{{Account: util.Uint160{1, 5, 9}}},
+				Scripts: []transaction.Witness{{
+					InvocationScript:   []byte{1, 4, 7},
+					VerificationScript: []byte{3, 6, 9},
+				}},
+			}
+			fallbackTx := &transaction.Transaction{
+				Script:          []byte{byte(opcode.RET)},
+				ValidUntilBlock: 123,
+				Attributes: []transaction.Attribute{
+					{Type: transaction.NotValidBeforeT, Value: &transaction.NotValidBefore{Height: 123}},
+					{Type: transaction.ConflictsT, Value: &transaction.Conflicts{Hash: mainTx.Hash()}},
+					{Type: transaction.NotaryAssistedT, Value: &transaction.NotaryAssisted{NKeys: 0}},
+				},
+				Signers: []transaction.Signer{{Account: util.Uint160{1, 4, 7}}, {Account: util.Uint160{9, 8, 7}}},
+				Scripts: []transaction.Witness{
+					{InvocationScript: append([]byte{byte(opcode.PUSHDATA1), keys.SignatureLen}, make([]byte, keys.SignatureLen)...), VerificationScript: make([]byte, 0)},
+					{InvocationScript: []byte{1, 2, 3}, VerificationScript: []byte{1, 2, 3}}},
+			}
+			p := &payload.P2PNotaryRequest{
+				MainTransaction:     mainTx,
+				FallbackTransaction: fallbackTx,
+				Witness: transaction.Witness{
+					InvocationScript:   []byte{1, 2, 3},
+					VerificationScript: []byte{7, 8, 9},
+				},
+			}
+			nrBytes, err := p.Bytes()
+			require.NoError(t, err)
+			str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(nrBytes))
+			submitNotaryRequest(t, true, neorpc.ErrVerificationFailedCode, str)(t)
+		})
+		t.Run("valid request", func(t *testing.T) {
+			sender := testchain.PrivateKeyByID(0) // owner of the deposit in testchain
+			notaryRequest = createValidNotaryRequest(chain, sender, 1)
+			nrBytes, err := notaryRequest.Bytes()
+			require.NoError(t, err)
+			str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(nrBytes))
+			submitNotaryRequest(t, false, 0, str)(t)
+		})
+	})
+
+	t.Run("getrawnotarypool", func(t *testing.T) {
+		t.Run("nonempty pool", func(t *testing.T) {
+			//get notary pool & check tx hashes
+			body := doRPCCallOverHTTP(rpcPool, httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+			actual := new(result.RawNotaryPool)
+			require.NoError(t, json.Unmarshal(res, actual))
+			require.Equal(t, 1, len(actual.Hashes))
+			for actMain, actFallbacks := range actual.Hashes {
+				require.Equal(t, notaryRequest.MainTransaction.Hash(), actMain)
+				require.Equal(t, 1, len(actFallbacks))
+				require.Equal(t, notaryRequest.FallbackTransaction.Hash(), actFallbacks[0])
+			}
+		})
+	})
```

Comment:
We also have the `getrawnotarypool` test above, squash this test with it.

Reply:
They are separated because the submitnotaryrequest occurred between them. This is why the "nonempty pool" is used. Currently, they are sharing the server, as you previously mentioned in https://github.com/nspcc-dev/neo-go/pull/3098#discussion_r1307298615

Reply:
What prevents you from adding yet another notary request in the `getrawnotarypool` test as a preparateion step in `nonemptypool` test?

Reply:
So it would be
```
t.Run("getrawnotarypool", func(t *testing.T) {
		t.Run("empty pool", func(t *testing.T) {...}
		t.Run("nonempty pool", func(t *testing.T) {// add notary request and do the `getrawnotarypool` afterwards ...}
})
```

Reply:
> They are separated because the submitnotaryrequest occurred between them

It's a separate thing, if `submitnotaryrequest` test fails than `nonemptypool` shouldn't depend on it.

Reply:
Fixed.